internal/cmd/gentypes/main.go

```diff
@@ -70,19 +70,44 @@ func run(args []string) error {
 
 func generateTypes(spec *btf.Spec) ([]byte, error) {
 	objName := &btf.Array{Nelems: 16, Type: &btf.Int{Encoding: btf.Char, Size: 1}}
+	mapID := &btf.Int{Size: 4}
+	progID := &btf.Int{Size: 4}
 	linkID := &btf.Int{Size: 4}
 	btfID := &btf.Int{Size: 4}
 	typeID := &btf.Int{Size: 4}
-	pointer := &btf.Int{Size: 8}
+	rawPointer := &btf.Int{Size: 8}
+	stringPointer := &btf.Int{Size: 8}
+	stringSlicePointer := &btf.Int{Size: 8}
+	bytePtr := &btf.Int{Size: 8}
```

Comment:
Do the target type sizes need to be correct here? It's a bit confusing regardless.

Reply:
The size needs to be equivalent to the size of a pointer. Using `Pointer` is a bit cumbersome because it requires specifying a target. I'll add a comment / try to work around this somehow.

Reply:
Turns out that using Pointer works just fine an is a bit clearer.